### PR TITLE
Feat/order

### PR DESCRIPTION
**Pull request by ntminh2107**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -13,14 +13,13 @@ const app = express()
...
   app.use(applyMiddlewares())
-
+  app.use(errorHandler)
   /*Health check */
   app.get('/healthcheck', (_, res) =>
     res.sendStatus(HttpStatusCode.ACCEPTED).json({ message: 'healthy' })
   )
 
   app.use('/api', getRouter())
-  app.use(errorHandler)
```
Comment (ahtdc22) [inline src/index.ts:23]: revert this file

**Comment by tung-dnt** [comment]
refactor all response, error status code, error msg

File: src/controllers/cart.controller.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    throw new HttpError(
+      'something wrong,pls try again',
+      HttpStatusCode.NOT_ALLOWED
+    )
+  const getCart = await getCartUser(userID)
+  res.status(HttpStatusCode.ACCEPTED).json({ message: 'cart detail', getCart })
+}
+
+const updateQuantityItm = async (req: Request, res: Response) => {
+  const userID = req.user?.id
```
Comment (tung-dnt) [inline src/controllers/cart.controller.ts]: const userID = req.user as User

File: src/services/auth.service.ts [hunk 3]
```diff
@@ -11,79 +11,87 @@ export const register = async (
...
-    .from(tblUser)
-    .innerJoin(tblRole, eq(tblUser.roleID, tblRole.id))
-    .where(eq(tblUser.email, email))
-    .limit(1)
+  return await db.transaction(async (trx) => {
+    const user = await trx
+      .select({
+        id: tblUser.id,
+        email: tblUser.email,
+        password: tblUser.password,
```
Comment (tung-dnt) [inline src/services/auth.service.ts]: remove

**Pull request merged**